core/src/test/scala/com/salesforce/op/ModelInsightsTest.scala

```diff
@@ -761,4 +774,29 @@ class ModelInsightsTest extends FlatSpec with PassengerSparkFixtureTest with Dou
     absError / bigCoeffSum < tol shouldBe true
     absError2 / smallCoeffSum < tol shouldBe true
   }
+
+  val tol2 = 0.1
+  val MomentsAndCard = getFeatureMomentsAndCard(standardizedLinpred, linRegDF._3)
+  val moments = MomentsAndCard._1
+  val cardinality = MomentsAndCard._2
+  val uniqueValsmallFeature = smallNorm.map(_.toDouble.get).toSet
+  val uniqueValbigFeature = bigNorm.map(_.toDouble.get).toSet
+
+  it should "correctly return moments calculation for numeric features" in {
+    moments.size shouldBe 2
+    moments(0).count shouldBe 1000
+    math.abs(moments(1).mean) < tol2 shouldBe true
+    math.abs(moments(1).variance - 10.0) / (moments(1).variance + 10.0)  < tol2 shouldBe true
+    moments(1).count shouldBe 1000
+    math.abs(moments(0).mean - 10000) / (moments(0).mean + 10000) < tol2 shouldBe true
```

Comment:
Also, you can compare to the actual mean of the sample rather than the mean of the distribution you're pulling from. That would let you use tighter tolerances and would probably be less subject to flakyness for small samples.

Reply:
good point, it was pretty flaky and i had to pick a higher threshold...